src/lightning_app/cli/cmd_clusters.py

```diff
@@ -0,0 +1,240 @@
+            V1ClusterType.BYOC: Text("byoc", style="bold yellow"),
+            V1ClusterType.GLOBAL: Text("lightning-cloud", style="bold green"),
+        }
+        for cluster in self.clusters:
+            cluster: Externalv1Cluster
+            status = phases[cluster.status.phase]
+            if cluster.spec.desired_state == V1ClusterState.DELETED and cluster.status.phase != V1ClusterState.DELETED:
+                status = Text("terminating", style="bold red")
+
+            # this guard is necessary only until 0.3.93 releases which includes the `created_at`
+            # field to the external API
+            created_at = datetime.now()
+            if hasattr(cluster, "created_at"):
+                created_at = cluster.created_at
+
+            table.add_row(
+                cluster.id,
+                cluster.name,
+                cluster_type_lookup.get(cluster.spec.cluster_type, Text("unknown", style="red")),
+                status,
+                created_at.strftime("%Y-%m-%d") if created_at else "",
+            )
+        return table
+
+
+def _wait_for_cluster_state(
+    api_client: LightningClient,
+    cluster_id: str,
+    target_state: V1ClusterState,
+    max_wait_time: int = MAX_CLUSTER_WAIT_TIME,
+    check_timeout: int = CLUSTER_STATE_CHECKING_TIMEOUT,
+):
+    """_wait_for_cluster_state waits until the provided cluster has reached a desired state, or failed.
+
+    Args:
+        api_client: LightningClient used for polling
+        cluster_id: Specifies the cluster to wait for
+        target_state: Specifies the desired state the target cluster needs to meet
+        max_wait_time: Maximum duration to wait (in seconds)
+        check_timeout: duration between polling for the cluster state (in seconds)
+    """
+    start = time.time()
+    elapsed = 0
+    while elapsed < max_wait_time:
+        cluster_resp = api_client.cluster_service_list_clusters()
+        new_cluster = None
+        for clust in cluster_resp.clusters:
+            if clust.id == cluster_id:
+                new_cluster = clust
+                break
+        if new_cluster is not None:
+            if new_cluster.status.phase == target_state:
+                break
+            elif new_cluster.status.phase == V1ClusterState.FAILED:
+                raise click.ClickException(f"Cluster {cluster_id} is in failed state.")
+            time.sleep(check_timeout)
+        elapsed = time.time() - start
+    else:
+        raise click.ClickException("Max wait time elapsed")
+
+
+def _check_cluster_name_is_valid(_ctx, _param, value):
+    pattern = r"^(?!-)[a-z0-9-]{1,63}(?<!-)$"
+    if not re.match(pattern, value):
+        raise click.ClickException(
+            """The cluster name is invalid.
+            Cluster names can only contain lowercase letters, numbers, and periodic hyphens ( - ).
+            Provide a cluster name using valid characters and try again."""
+        )
+    return value
+
+
+_default_instance_types = frozenset(
```

Comment:
As discussed with Karolis, we can't have any references to the instance types in the python framework. The backend decides which resources are available or what their names are, and any validation regarding that happens in the backend. This is the same with CloudCompute, where we don't do any validation to its input nor list any defaults. 

So I'm pretty sure we need to remove this list.

Reply:
cc: @rusenask

Reply:
dropped in a follow up commit - we'll populate this in the backend if `None` are provided.